xbmc/cores/AudioEngine/Sinks/AESinkDARWINOSX.cpp

```diff
@@ -93,13 +93,56 @@ static CADeviceList GetDevices()
   return list;
 }
 
+//---------- DEVICE CHANGED EVENT HANDLING
+// when device list change callbacks are received
+// inform the ae engine this amount of time after
+// the last callback was received (bundle multiple callbacks)
+// into one final call
+#define DEVICE_LIST_CHANGED_TIMER_MS 1500
+
+CDeviceChangeListTimeout::CDeviceChangeListTimeout()
+{
+  m_pDeviceListChangedTimer = new CTimer(this);
```

Comment:
In AddPackets - not nice - but that's the only thing that does something (at least if the sink is active). If it is not -> it is suspended and unloaded anyways.